library/psa_crypto_cipher.c

```diff
@@ -482,16 +485,17 @@ static psa_status_t cipher_encrypt( const psa_key_attributes_t *attributes,
                             output + operation.iv_length,
                             output_size - operation.iv_length,
                             &olength );
-    accumulated_length += olength;
     if( status != PSA_SUCCESS )
         goto exit;
 
+    accumulated_length += olength;
+
     status = cipher_finish( &operation, output + accumulated_length,
                             output_size - accumulated_length, &olength );
-    accumulated_length += olength;
     if( status != PSA_SUCCESS )
         goto exit;
 
+    accumulated_length += olength;
```

Comment:
```suggestion
    *output_length = accumulated_length + olength;
```
That's probably a bit better.